app/src/lib/find-default-branch.ts

```diff
@@ -1,8 +1,23 @@
 import { Branch, BranchType } from '../models/branch'
-import { Repository } from '../models/repository'
+import { Repository, getForkContributionTarget } from '../models/repository'
+import { ForkContributionTarget } from '../models/workflow-preferences'
 import { getRemoteHEAD } from './git'
 import { getDefaultBranch } from './helpers/default-branch'
 
+async function getContributionTargetHead(
+  repository: Repository
+): Promise<[string, string | null]> {
+  if (getForkContributionTarget(repository) === ForkContributionTarget.Parent) {
+    const head = await getRemoteHEAD(repository, 'upstream')
+
+    if (head !== null) {
+      return ['upstream', head]
+    }
```

Comment:
Shouldn't we always return here? The calling method will then set the `remoteRef` to `undefined` if there was no `head`. I _think_ it makes sense to stick to the `upstream` HEAD if the fork contribution target is the parent repo, even if for whatever reason we can't find its HEAD. Does that make sense?
```suggestion
    return ['upstream', head]
```

Reply:
Yep, it sure does. I took another stab this and at the same time addressed the (very rare) case when the default remote isn't named `origin`.